demo/dlb-init.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+enable_and_configure_vfs() {
+  DEVPATH=$1
+  NUM_VFS_PER_PF=$2
+  # enable sriov
+  SRIOV_NUMVFS_PATH="$DEVPATH"/sriov_numvfs
+  if test -w "$SRIOV_NUMVFS_PATH" -a "$(cat "$SRIOV_NUMVFS_PATH")" -eq 0 ; then  
+    echo "$NUM_VFS_PER_PF" | tee -a "$SRIOV_NUMVFS_PATH"
```

Comment:
What is the idea behind `echo |tee -a` used here and below? This script will print set of numbers to stdout, and same values are appended to target files. Is stdout used for something, or is it just for debug? For debug, I would recommend adding -x to bash line at top (can be removed in production). 
About `tee -a`: This will append to file, means if you run this multiple times, you will have multiple entries in those files. Is this intentional, and does it make sense? I recall the related bmaas-script had simple redirection echo $num > $file, cant this method be used here as well?

Reply:
I think this syntax is a leftover from scripts where sudo is used:

```
$ sudo echo 0000:50:00.0 > /sys/bus/pci/drivers/c6xx/unbind 
-bash: /sys/bus/pci/drivers/c6xx/unbind: Permission denied
$ echo 0000:50:00.0 | sudo tee /sys/bus/pci/drivers/c6xx/unbind 
0000:50:00.0
```

> echo $num > $file, cant this method be used here as well?

I don't see why not.

Reply:
@mythi Actually true.! It is leftover from the code when we put 'sudo' there. But, I did not see any significant difference and kept it. If there are differences between `echo | tee -a` and `echo $num > $file`, let me change to that.! Thank you!

Reply:
Difference is that we load and execute tee binary totally without any gain number of times, and (although we are not pressed for resources) IMHO this is not the way to run programs if there is better way. By same logic, one can invent even longer piped chain where end result will be same and claim "use it as end result is same". We should develop towards shortest, lightest and cleanest.

Reply:
Also, running less processes has definitely measurable nano-effect to reduce climate change rate! (although my writing about it here drags it the other way a bit :-)

Reply:
@okartau I remembered another reason why I used `echo | tee`. In toybox, it complains like this.
> echo: write: Invalid argument
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
echo: write: Bad file descriptor
/sys/devices/pci0000:72/0000:72:00.0echo: write: Bad file descriptor

Reply:
@okartau Thanks for the explanations.!!

Reply:
@okartau Ahaha, yes true.! Sorry for the bad effect to the earth.